cli/azd/cmd/middleware/command_hooks.go

```diff
@@ -0,0 +1,87 @@
+package middleware
+
+import (
+	"context"
+	"log"
+
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/exec"
+	"github.com/azure/azure-dev/cli/azd/pkg/ext"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+)
+
+func UseCommandHooks(
+	commandOptions *internal.GlobalCommandOptions,
+	console input.Console,
+	commandRunner exec.CommandRunner,
+) actions.MiddlewareFn {
+	return func(ctx context.Context, options *actions.ActionOptions, next actions.NextFn) (*actions.ActionResult, error) {
+		azdContext, err := azdcontext.NewAzdContext()
+		if err != nil {
+			log.Printf("failing creating AzdContext for command hooks, %s", err.Error())
+			return nil, err
+		}
+
+		environmentName := &commandOptions.EnvironmentName
```

Comment:
Shoot. I had missed out on removing this in #1143 . Sending out #1230, but this needs to be replaced.

Reply:
ok - will try to figure something out.

Reply:
I introduced a new private interface that all flag sets that support environment name so that I can lazily cast the flags during dependency injection.